### ISSUE TITLE
Conditional Version Pattern

### ISSUE THREAD

**Issue by cecom** [jgitver/jgitver#81]
Hi,

this feature request is something which is a little bit discussed in #33 and in #76. The solution of #76 looks good, but i want to go a step further. We use the [vnie Branching](https://nvie.com/posts/a-successful-git-branching-model/) model and we want to create tags when a branch is finished, not when the branch is started. So our calculation of the version is a little bit different. Here some examples:

Number | Branch | JGITVER Example Result | JGITVER Pseudo Rule
------------ | ------------ | ------------- | -------------
1 | master        | 1.0.0     | Use the latest TAG
2 | develop       | develop-54 | No TAG usage, count the commits from start of project
3 | release/1.0.0 | 1.0.0-52 | No TAG usage, extract ${M}.${m}.${p} from Branch name and add commits from start of project 
4 | feature/x     | feature_x-54 | No TAG usage, count the commits from start of project

Basically we want #76 with the addition of:
- Configure to calcuclate the ${M}.${m}.${p} from the Branchname --> we could use a transformation or take it as is e.g.:
```
<branchPolicy>
   <pattern>release\/(.*)</pattern>
   <transformations>
      <transformation>EXTRACT_VERSIONNUMBER</transformation>
   </transformations>
   <versionPattern>${~meta.QUALIFIED_BRANCH_NAME}${-alpha~meta.COMMIT_DISTANCE_FROM_START}-${meta.GIT_SHA1_8}</versionPattern>
</branchPolicy>
```
- add a meta meta.COMMIT_DISTANCE_FROM_START
